### PR TITLE
[cmake] Enable building tests when cross compiling

### PR DESCRIPTION
**Pull request by antiagainst**
This commit enables -DIREE_BUILD_TESTS=ON when invoking CMake
with a toolchain file. Now we can compile non-host-only tests
towards, e.g., ARM for Android.

This commit also enables cross-compiling iree-benchmark-module.

### CODE REVIEW

File: docs/GetStarted/cmake_options_and_variables.md [hunk 1]
```diff
@@ -114,6 +114,32 @@ CMake cannot handle multiple compiler toolchains in one CMake invocation. So the
...
+
+The tricky part is to set up target artifacts' dependencies on host executables.
+Normally in CMake we define targets and `target_link_libraries` to express the
+dependency relationship between targets. The optimal way is to [`export`](https://cmake.org/cmake/help/latest/command/export.html)
+targets under host configuration and then [`import`](https://cmake.org/cmake/help/latest/command/add_executable.html?highlight=import#imported-executables)
+into target configuration. But that is not playing well with LLVM's TableGen
+configurations. LLVM's cross-compilation uses on files for dependencies. So
+here we need to do the same. This results in reverting the relationship between
+IREE artifact output names (e.g., `iree-translate`) and IREE package-prefixed
+CMake target names (e.g., `iree_tools_iree-translate`).
```
Comment (ScottTodd) [inline docs/GetStarted/cmake_options_and_variables.md:132]: Good details, but a bit worried that they are more developer-oriented (for anyone maintaining IREE's CMake macros themselves).

File: docs/GetStarted/getting_started_android_cmake.md [hunk 2]
```diff
@@ -82,7 +82,7 @@ $ cmake -G Ninja -B build-android  \
     for your target device. You can also refer to Android NDK's
     [CMake documentation](https://developer.android.com/ndk/guides/cmake) for
     more toolchain arguments.
-*   Building IREE compilers, tests, and samples for Android is not supported at
+*   Building IREE compilers and samples for Android is not supported at
     the moment; they will be enabled soon.
```
Comment (ScottTodd) [inline docs/GetStarted/getting_started_android_cmake.md:86]: Do we actually plan on getting the compiler (and LLVM) building for Android? Are we going to use that?

File: docs/GetStarted/cmake_options_and_variables.md (hunk 1 above)
Comment (antiagainst) [reply docs/GetStarted/cmake_options_and_variables.md:132]: Yup. I feel we need another doc to describe the logic behind build systems. Tracked by https://github.com/google/iree/issues/2111. I'll add a TODO here to move after we have such a doc.

File: docs/GetStarted/getting_started_android_cmake.md (hunk 2 above)
Comment (antiagainst) [reply docs/GetStarted/getting_started_android_cmake.md:86]: Maybe. :) Not immediately for IREE; but I can envision that projects might want to reuse IREE compilers to do online conversion. :)

**Pull request merged**